packages/labs/ssr/src/test/lib/render-lit_test.ts

```diff
@@ -545,6 +552,133 @@ for (const global of [emptyVmGlobal, shimmedVmGlobal]) {
       );
     }
   });
+
+  test('server-only templates', async () => {
+    const {render, trivialServerOnly} = await setup();
+    const result = await render(trivialServerOnly);
+    assert.is(result, `<div>Server only</div>`);
+  });
+
+  test('server-only template with a binding', async () => {
+    const {render, serverOnlyWithBinding} = await setup();
+    const result = await render(serverOnlyWithBinding);
+    assert.is(result, `<div>Server only</div>`);
+  });
+
+  test('server-only template with an array', async () => {
+    const {render, serverOnlyArray} = await setup();
+    const result = await render(serverOnlyArray);
+    assert.is(result, `<div>onetwothree</div>`);
+  });
+
+  test('server-only template inside server-only template', async () => {
+    const {render, serverOnlyInsideServerOnly} = await setup();
+    const result = await render(serverOnlyInsideServerOnly);
+    assert.is(result, `<div>Server only</div>`);
+  });
+
+  test(`server-only template can't render a normal template`, async () => {
+    const {render, serverOnlyRenderHydratable} = await setup();
+    const result = render(serverOnlyRenderHydratable);
+    assert.is(
+      result,
+      `
+    <div>server only</div>
+    <!--lit-part AEmR7W+R0Ak=--><div><!--lit-part-->hydratable<!--/lit-part--></div><!--/lit-part-->
+  `
+    );
+  });
+
+  test(`normal template can't render a server-only template`, async () => {
+    const {render, hydratableRenderServerOnly} = await setup();
+    assert.throws(
+      () => render(hydratableRenderServerOnly),
+      /A server-only template can't be rendered inside an ordinary, hydratable template./
+    );
+  });
+
+  test('server-only template into raw element', async () => {
+    const {render, serverOnlyRawElementTemplate} = await setup();
+    const result = await render(serverOnlyRawElementTemplate);
+    assert.is(
+      result,
+      `
+    <title>No comments inside</title>
+    <textarea>This also works (kinda).</textarea>
```

Comment:
Why `(kinda)`?

Reply:
Ah, good point. It works with the full promise of server-only templates. You can render them once, but you can't update them!